### ISSUE TITLE
Aliases and Qualified name mapping

### ISSUE THREAD

**Issue by bencassie** [roadrunnerlenny/etlbox#29]
Hi there,

Discovered this project the other day and am really impressed with it. However, I have a question with regards to selecting data from the database.

If i write a DBSource select statement such as

```sql
select id as AssetId, secid as SecurityID
from asset_table
```

The DBSource hits an issue with the tsql parser identifying 4 columns rather than just the 2. The exception raised is indexOutOfBounds when reading the non-existent columns.

Similarly if I write something like
```sql
select ass.Id, sec.Id
from asset ass
inner join security sec on ass.id = sec.id
```

I get the same issue as the table qualifier is interpreted as an identifier token.

Is there any kind of work around at all?

Many thanks

**Comment by roadrunnerlenny** [comment]
Hi, 

thanks for using ETLBox and reporting a bug. 

In my current tests where I use Sql statements as data source, I wrote only tests with statements that were like "select col1, col2, col3, ... from table" - but I did not write tests using more sophisticated statements. But my goal was definitively to support more sophisticated statements  as well, so good that you found the bug here. 

I just wrote some tests and reproduced the issue. Right now I can't come up with a bugfix - it's not so easy to fix as statements can become kind of complex and the T-SQL parser library I am currently using seems to have some limitations here. 

In the meanwhile, I can offer you a workaround:
You can define a TableDefinition yourself and pass it to the source. E.g, for your first example, your code would look like this:

```
var TableDefinition = new TableDefinition("asset_table"
                , new List<TableColumn>() {
                new TableColumn("AssetId", "INT"),
                new TableColumn("SecurityId", "INT")
            });

var DBSource<type> = new DBSource<type() {
  Sql = "select id as AssetId, secid as SecurityID from asset_table",
  SourceTableDefinition = TableDefinition
}
```

As you can see, you pass the name and datatype of the column that your sql would return. 
If you pass a TableDefinition, the DBSource won't try to parse this from the statement itself. This should help you out here.

**Comment by bencassie** [comment]
Thank you very much for your response, that works for me.

**Comment by roadrunnerlenny** [comment]
Hi,
I just released version 1.6.2 (https://www.nuget.org/packages/ETLBox/1.6.2) which includes an improved Sql parser. Now the column names should be parsed from the Sql statement without error. That means if you use version 1.6.2 you don't have to pass a `TableDefinition` object anymore.
